src/test/eo/org/eolang/collections/list-tests.eo

```diff
@@ -601,3 +601,81 @@
           sprintf
             "%d\n"
             x
+
+[] > filteredi-test-1
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 3 1 4 2 5
+        [v i]
+          v.lt 3 > @
+      list
+        * 1 2
+    $.equal-to TRUE
+
+[] > filteredi-test-2
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 7 3 8 4 2 1
+        [v i]
+          v.gt 3 > @
+      list
+        * 7 8 4
+    $.equal-to TRUE
+
+[] > filteredi-test-3
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          *
+        [v i]
+          v.lt 3 > @
+      list
+        *
+    $.equal-to TRUE
+
+[] > filteredi-test-4
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 3 1 4 2 5
+        [v i]
+          i.gt 3 > @
+      list
+        * 5
+    $.equal-to TRUE
+
+[] > filteredi-test-5
+  assert-that > @
+    eq.
+      filteredi.
+        list
+          * 3 1 4 2 5
+        [v i]
+          v.eq 0 > @
+      list
+        *
+    $.equal-to TRUE
+
+# This code should prints "1 2"
+[] > filteredi-printing-test
+  seq > @
+    filteredi.
+      list
+        * 1 2 3
+      [v i]
+        seq > @
+          if.
+            v.lt 3
+            []
```

Comment:
@ivan-egorov42 Why do we need this abstract object here?

Reply:
Because `if.` returns this abstract object if `v.lt 3` and `seq` will dataize it. You mean there should be another implementation for printing?

Reply:
@ivan-egorov42 I mean:
```
if.
  v.lt 3
  stdout 
    sprintf
      "%d\n"
       v
  FALSE
```